apiserver/facades/agent/deployer/deployer.go

```diff
@@ -36,14 +37,16 @@ type ApplicationService interface {
 	GetUnitLife(context.Context, coreunit.Name) (life.Value, error)
 	EnsureUnitDead(context.Context, coreunit.Name, leadership.Revoker) error
 	RemoveUnit(context.Context, coreunit.Name, leadership.Revoker) error
+	GetUnitWorkloadStatus(context.Context, coreunit.Name) (*corestatus.StatusInfo, error)
+	SetUnitWorkloadStatus(context.Context, coreunit.Name, *corestatus.StatusInfo) error
 }
 
 // DeployerAPI provides access to the Deployer API facade.
 type DeployerAPI struct {
 	*common.PasswordChanger
 	*common.APIAddresser
 	*common.UnitsWatcher
-	*common.StatusSetter
+	*common.UnitStatusSetter
```

Comment:
I don't want to compose these anymore. Instead, I want to add them as members. That means copying the method signatures and direct them to the member. This means that if someone changes the common type, it doesn't break the API server contract.

Reply:
Yes agree. Done